templates/install_puppet.bat.erb

```diff
@@ -40,6 +40,34 @@ REM This may fail on agents without pxp-agent, but since this is not
 REM run interactively and the next command sets ERRORLEVEL, it's OK.
 net stop pxp-agent
 
+<% if @msi_move_locked_files %>
+REM Move puppetres.dll to avoid file locks and service restarts (MODULES-4207
+)
+REM Puppet-Agent, less than 1.6.0, location
+SET PUPPETRES=<%= @env_windows_installdir %>\misc\puppetres.dll
+SET RANDFILE=%TEMP%\puppetres-%RANDOM%-1.DLL

```

Comment:
Because if we can't move a file to a name that already exists.  If the file we're moving to exists and cannot be deleted we're in a bad situation.  In this way we mostly guarantee that there will be no existing file with that name.